storage_controller/src/compute_hook.rs

```diff
@@ -608,13 +639,19 @@ impl ComputeHook {
     /// if something failed.
     pub(super) fn notify_background(
         self: &Arc<Self>,
-        notifications: Vec<(TenantShardId, NodeId, ShardStripeSize)>,
+        notifications: Vec<(
+            TenantShardId,
+            NodeId,
+            ShardStripeSize,
+            Option<AvailabilityZone>,
+        )>,
```

Comment:
Done in 670cdefa04c455fd9d7625590166aac456eac19b -- this was something I'd avoided earlier because in some contexts we can use refs for AZ, but I just bit the bullet and made this a Cow.